verifier/newverifier.go

```diff
@@ -0,0 +1,45 @@
+package verifier
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/spacemeshos/merkle-tree"
+	"github.com/spacemeshos/poet/shared"
+	"sort"
+)
+
+func Validate(proof shared.MerkleProof, challenge shared.NewChallenge, leafCount uint64, securityParam uint8) error {
+	if int(securityParam) != len(proof.ProvenLeaves) {
+		return fmt.Errorf("number of proven leaves (%d) must be equal to security param (%d)",
+			len(proof.ProvenLeaves), securityParam)
+	}
+	provenLeafIndices := asSortedSlice(shared.FiatShamir(proof.Root, leafCount, securityParam))
```

Comment:
The pattern of calling FiatShamir and then sorting the result repeats itself. Consider wrapping it in a single function (will also prevent bugs where people forget to sort the result)

Reply:
if all usages of this function would require a sorted list, I'd agree. However, some need the raw output (`map[uint64]bool`) which is why I've created a function to transform it into a sorted slice.

Since the data types are different, there's no risk of someone forgetting to run the wrapping function.

Reply:
All usages of the Fiat Shamir method are also sorted. I see no real reason to use a map and not a slice